napari/layers/base/base.py

```diff
@@ -1020,6 +1025,16 @@ def _type_string(self) -> str:
     def as_layer_data_tuple(self):
         state = self._get_state()
         state.pop('data', None)
+        if hasattr(self, '_deprecated_properties_list'):
+            state = DeprecatingDict(state)
+            element: DeprecatedProperty
+            for element in self._deprecated_properties_list:
```

Comment:
```suggestion
            for element in self._deprecated_properties_list:
                if element.new_name not in state:
                    continue
```

@andy-sweet I think that it should solve the problem.

Reply:
Yes, I have something similar.